healsparse/geom.py

```diff
@@ -0,0 +1,668 @@
+        else:
+            v = value[i]
+
+        circle = Circle(ra=ra[i], dec=dec[i], radius=radius[i], value=v)
+        circles.append(circle)
+
+    return circles
+
+
+def realize_geom(geom, smap, type='or'):
+    """
+    Realize geometry objects in a map
+
+    Parameters
+    ----------
+    geom: geometric primitive or list thereof
+        List of Geom objects, e.g. Circle, Polygon
+    smap: HealSparseMaps
+        The map in which to realize the objects
+    """
+
+    if type != 'or':
+        raise ValueError('type of composition must be or')
+
+    if not smap.isIntegerMap:
+        raise ValueError('can only or geometry objects into an integer map')
+
+    if not isinstance(geom, (list, tuple)):
+        geom = [geom]
+
+    # split the geom objects up by value
+    gdict = {}
+    for g in geom:
+        value = g.value
+        _check_int(value)
+        _check_int_size(value, smap.dtype)
+
+        if value not in gdict:
+            gdict[value] = [g]
+        else:
+            gdict[value].append(g)
+
+    # deal with each value separately and add to
+    # the map
+    for value, glist in gdict.items():
+        for i, g in enumerate(glist):
+            tpixels = g.get_pixels(nside=smap.nsideSparse)
+            if i == 0:
+                pixels = tpixels
+            else:
+                oldsize = pixels.size
+                newsize = oldsize + tpixels.size
+                # need refcheck=False because it will fail when running
+                # the python profiler; I infer that the profiler holds
+                # a reference to the objects
+                pixels.resize(newsize, refcheck=False)
+                pixels[oldsize:] = tpixels
+
+        pixels = np.unique(pixels)
+
+        values = smap.getValuePixel(pixels)
+        values |= value
+        smap.updateValues(pixels, values)
+
+
+def _check_int(x):
+    check = (
+        issubclass(x.__class__, np.integer) or
+        issubclass(x.__class__, int)
+    )
+    if not check:
+        raise ValueError('value must be integer type, '
+                         'got %s' % x)
+
+
+def _check_int_size(value, dtype):
+    ii = np.iinfo(dtype)
+    if value < ii.min or value > ii.max:
+        raise ValueError('value %d outside range [%d, %d]' %
+                         (value, ii.min, ii.max))
+
+
+class GeomBase(object):
+    """
+    base class for goemetric objects that can convert
+    themselves to maps
+    """
+
+    @property
+    def value(self):
+        """
+        get the value to be used for all pixels in the map
+        """
+        return self._value
+
+    def get_pixels(self, *, nside):
+        """
+        get pixels for this map
+
+        Parameters
+        ----------
+        nside: int
+            Nside for the pixels
+        """
+        raise NotImplementedError('implment get_pixels')
+
+    def get_map(self, *, nside, dtype):
+        """
+        get a healsparse map corresponding to this geometric primitive
+
```

Comment:
Is this really used in practice, or just in testing?  Because I think that `realize_geom()` is the main way to make a map, and when I was playing around with this I was confused about which I should use.

Reply:
yes, it is useful because it is so easy to use compared to constructing a map and adding to it

especially nice at the command line